### PR TITLE
stop recording if necessary while a long press

### PR DESCRIPTION
**Pull request by lwalter86**
Hi,
Recording function and led blinking task were not stopped when we did a long press on the button.
king regards,
lionel

### CODE REVIEW

File: pingo_parts_pushbutton_long_short.py [hunk 1]
```diff
@@ -144,19 +144,35 @@ def _update_led(self):
 
     def on_push_button_pressed_long(self):
         logger.info("push button pressed long")
+        
+        # In case we stop the system even when we're still recording
+        if self.recording:
+            self.led.stop() # stop blinking
+            self.stop_recording()
+            self.recording = False
+            time.sleep(0.5)  # wait the thread closure. Need to be reviewed
```
Comment (s-celles) [inline pingo_parts_pushbutton_long_short.py:153]: This (`time.sleep(0.5)`) can probably be improved

**Comment by s-celles** [comment]
Merging anyway

**Comment by lwalter86** [comment]
`is_alive()` function in threading.Thread class ?

Edit: using ` for inline code

**Comment by s-celles** [comment]
Maybe if you are considering using `is_alive` method you need to access `BlinkTask` thread.

see https://github.com/pingo-io/pingo-py/blob/master/pingo/parts/led.py#L61

so if you want to do this I will take `Led` class and `BlinkTask`

I will modify BlinkTask and inherit from `threading.Thread`.

so instead of

```
    self.blink_task = BlinkTask(self, times, on_delay, off_delay)
    threading.Thread(target=self.blink_task.run).start()
```

we will have

```
    self.blink_task = BlinkTask(self, times, on_delay, off_delay)
    self.blink_task.start()
```

I will put `is_alive` in https://github.com/pingo-io/pingo-py/blob/master/pingo/parts/led.py#L59

**Pull request merged**